dts/bindings/serial/nordic,nrf-uarte.yaml

```diff
@@ -14,3 +14,9 @@ properties:
     type: boolean
     description: |
       UARTE has RX frame timeout HW feature.
+  pin-retention:
+    type: boolean
+    description: |
+      UARTE pin states must be retained when peripheral is disabled.
+      It is required when CTRLSEL is used to route UARTE pins. It is always the
+      case for fast peripheral and in special cases for slow peripherals.
```

Comment:
this can be inferred from the power domain UART resides?

Reply:
The only thing that comes to my mind is to add build assert in code where there are fixed location checked but we try to avoid putting soc specific stuff in the code.

Maybe we could come up with a way of describing that in DT. e.g. create a description of pool of available pins with some specific properties that are required for a given pool. This could be useful not only in this case but also in case of Lumos where peripherals are fixed to a single gpio port.

Reply:
We should likely review the pinctrl model we have, as new socs have changed it. For example, 54H20 has a mix of PSEL (peripheral-level) and CTRLSEL (classical multiplexer).